docs/AUTHENTICATION.md

```diff
@@ -29,17 +33,85 @@ CUBEJS_JWT_ALGS=RS256
+   3. Make sure to keep the `Generate client secret` option checked.
+   4. Keep the default `Auth Flows Configuration` values selected.
+   5. Security configuration option should be `Enabled`.
+   6. Click on Create app client.
+   7. Note the `App client id` and `App client secret`. These will be used in later steps.
+8.  Choose `Resource servers` from the navigation bar on the left-side of the console page.
+   1.  Set the name of your resource server. For example: `Plio Staging Analytics Resource server`
+   2.  Set the identifier to the `App client id` you had noted in previous step.
+   3.  Create the following scopes:
+       1.  Name: `post`, Description: `post requests`
+       2.  Name: `get`, Description: `get requests`
+   4.  Create the resource server by clicking on save changes.
+9.  Choose `App client settings` from the navigation bar on the left-side of the console page.
+    1.  Select Cognito User Pool as one of the Enabled Identity Providers.
+    2.  Enter the callback URL and signout urls.
+    3.  Check the `client credentials` option under Allowed Oauth Flows.
+    4.  Select the custom scopes for `<client-id>/get` and `<client-id>/post` that you created in previous step.
+    5.  Click on save changes.
+10. Choose `Domain name` from the navigation bar on the left-side of the console page.
+    1.  Enter the domain prefix and click on save.
+    2.  The complete url will be used to generate access tokens.
+    3.  Note the url.
+11. To test your set up, run the following command in your shell. Make sure to update the values within `<>` as per your settings. It should return a valid response with access token, expiry time and token type.
+```sh
+curl --location --request POST \
+'https://<DOMAIN-URL>/oauth2/token?grant_type=client_credentials&client_id=<APP-CLIENT-ID>&client_secret=<APP-CLIENT-SECRET>' \
+--header 'Content-Type: application/x-www-form-urlencoded'
+```
+12. Make sure you have updated the Plio backend repository [environment variables](https://github.com/avantifellows/plio-backend/blob/master/docs/ENV.md#identity-provider-for-plio-analytics) with the Cognito app settings.
+
+For more details on Cognito configuration, check out the [official documentation from Cube.js](https://cube.dev/docs/security/jwt/aws-cognito).
+
+#### Configuring CUBE_JS environment variables for AWS Cognito
+Add the following environment variables for this CubeJS project after replacing the `<>` parts with your configs.
+```sh
+CUBEJS_JWK_URL=https://cognito-idp.ap-south-1.amazonaws.com/<USER-POOL-ID>/.well-known/jwks.json
+CUBEJS_JWT_ISSUER=https://cognito-idp.ap-south-1.amazonaws.com/<USER-POOL-ID>
+CUBEJS_JWT_ALGS=RS256
+CUBEJS_JWT_CLAIMS_NAMESPACE=<CLAIMS_NAMESPACE>
+```
+
+##### `CUBEJS_JWK_URL`
+This is the url to fetch the JSON web keys details. Replace the `<USER-POOL-ID>` by the pool id of your user pool. This can be determined from `Your User Pool > General Settings > Pool Id`.
```

Comment:
```suggestion
This is the url to fetch the JSON web key details. Replace the `<USER-POOL-ID>` by the pool id of your user pool. This can be determined from `Your User Pool > General Settings > Pool Id`.
```